.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,146 @@
+    if: github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.ref }}
+
+      - name: Validate release precondition
+        env:
+          RELEASE_VERSION: ${{ inputs.version }}
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          read -r release type < <(gh release list | awk -v release="v$RELEASE_VERSION" '$1 ~ release { print $1,$2; ++n } END { if (n == 0) print "undefined", "undefined" }')
+          if [[ "$release" == "undefined" ]]; then
+            echo "Release v$RELEASE_VERSION does not exist. Proceeding"
+            echo "create_draft_release=true" >> "$GITHUB_ENV"
+          else
+            if [[ "$type" != "Draft" ]]; then
+              echo "Release '$release' already exists and is not a draft, but has release state '$type'. Cannot proceed"
+              exit 1
+            else
+              echo "Release '$release' already exists and is a draft. Proceeding"
+              echo "create_draft_release=false" >> "$GITHUB_ENV"
+            fi
+          fi
+
+          release_pr=$(gh pr view rc/$RELEASE_VERSION --json title,state,number)
+
+          if [[ ! -z "$release_pr" ]]; then
+
+            pr_title=$(echo "$release_pr" | jq -r '.title')
+            pr_state=$(echo "$release_pr" | jq -r '.state')
+            pr_number=$(echo "$release_pr" | jq -r '.number')
+
+            echo "Found PR '$pr_title' with state '$pr_state'"
+
+            if [[ "$pr_title" == "Release v$RELEASE_VERSION" ]] && [[ "$pr_state" != "CLOSED" ]]; then
+              echo "Release PR is not closed, deleting it to proceed"
+              gh pr close --delete-branch $pr_number
+            fi
+          fi
+
+          if [[ ! -z $(git ls-remote --heads origin rc/$RELEASE_VERSION) ]]; then
+            echo "Deleting existing release branch"
+            git push origin --delete rc/$RELEASE_VERSION
+          fi
+
+          if [[ ! -z $(git ls-remote --heads origin feature/update-user-manual-for-$RELEASE_VERSION) ]]; then
+            echo "Deleting existing feature branch"
+            git push origin --delete feature/update-user-manual-for-$RELEASE_VERSION
+          fi
+
+      - name: Configure git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.9"
+
+      - name: Install release script dependencies
+        run: pip install -r scripts/release/requirements.txt
+
+      - name: Validate version
+        run: |
+          python scripts/release/validate-version.py "$RELEASE_VERSION"
+
+      - name: Create release branch
+        run: |
+          git switch -c rc/$RELEASE_VERSION
+          git push --set-upstream origin rc/$RELEASE_VERSION
+
+      - name: Create draft release
```

Comment:
We already have a script here for creating a draft release (`scripts/release/create_draft_release.sh`). That should be reviewed, incorporated and possibly deleted if it is no longer used.

Reply:
I will see if that could be incorporated or if this subsumes that script.

Reply:
I've removed the script and corresponding workflow.
Both are subsumed by the new release automation.